### PR TITLE
Replace "lvalue" terminology with "place".

### PR DESCRIPTION
**Pull request by eddyb**
See #46425 for the previous PR (which only changed MIR-related code).

r? @nikomatsakis

### CODE REVIEW

File: src/test/ui/issue-26093.stderr [hunk 1]
```diff
@@ -4,7 +4,7 @@ error[E0070]: invalid left-hand side expression
 13 |         $thing = 42;
    |         ^^^^^^^^^^^ left-hand of expression not valid
 ...
-19 |     not_an_lvalue!(99);
+19 |     not_a_place!(99);
    |     ------------------- in this macro invocation
```
Comment (kennytm) [inline src/test/ui/issue-26093.stderr]: This line needs to be shortened.

```
[00:45:32] ---- [ui] ui/issue-26093.rs stdout ----
[00:45:32] 	diff of stderr:
[00:45:32] 
[00:45:32] 5	   |         ^^^^^^^^^^^ left-hand of expression not valid
[00:45:32] 6	...
[00:45:32] 7	19 |     not_a_place!(99);
[00:45:32] +	   |     ----------------- in this macro invocation
[00:45:32] -	   |     ------------------- in this macro invocation
[00:45:32] 9	
[00:45:32] 10	error: aborting due to previous error
[00:45:32] 11	
```

Comment (eddyb) [reply src/test/ui/issue-26093.stderr]: Oh right, I'm not running `ui` tests locally because they were broken at some point...

**Review by nikomatsakis — APPROVED** [review]

**Pull request merged**